src/details/ArborX_DetailsAlgorithms.hpp

```diff
@@ -208,6 +209,9 @@ Point returnCentroid(Box const &box)
 KOKKOS_INLINE_FUNCTION
 Point returnCentroid(Sphere const &sphere) { return sphere.centroid(); }
 
+KOKKOS_INLINE_FUNCTION
+Point returnCentroid(Ray const &ray) { return ray.origin(); }
```

Comment:
Make a commit with the changes to that file saying something like "Add Details::returnCentroid(Ray) implementation to enable interacts(Ray) predicate in batched queries"